packages/w3up-client/src/client.js

```diff
@@ -291,9 +292,66 @@ export class Client extends Base {
         )
       }
     }
+
     return space
   }
 
+  /**
+   * Authorizes a gateway to serve content from the provided space and record egress events.
+   * Delegates the following capabilities to the gateway:
+   * - `space/content/serve/*`
+   *
+   * @param {import('./types.js').OwnedSpace} space - The space to authorize the gateway for.
+   * @param {object} [options] - Options for the authorization.
+   * @param {`did:web:${string}`} [options.gateway] - The Web DID of the gateway to authorize. If not provided, the default `did:web:w3s.link` gateway will be authorized.
+   * @param {number} [options.expiration] - The time in seconds to expire the authorization.
+   * @returns {Promise<import('./delegation.js').AgentDelegation<any>>} Resolves with the AgentDelegation instance once the gateway is successfully authorized.
+   */
+  async authorizeGateway(space, options = {}) {
+    const currentSpace = this.currentSpace()
+    try {
+      // Set the current space to the space we are authorizing the gateway for
+      await this.setCurrentSpace(space.did())
+
+      const authProof = await space.createAuthorization(this.agent)
```

Comment:
You shouldn't need to do this, you have this proof in your store already. This should only be required when you first create a spece.